src/snowflake/snowpark/_internal/analyzer/expression.py

```diff
@@ -225,6 +225,47 @@ def __init__(self, value: Any, datatype: Optional[DataType] = None) -> None:
             self.datatype = infer_type(value)
 
 
+class Interval(Expression):
+    def __init__(
+        self,
+        year: int = 0,
+        quarter: int = 0,
+        month: int = 0,
+        week: int = 0,
+        day: int = 0,
+        hour: int = 0,
+        minute: int = 0,
+        second: int = 0,
+        millisecond: int = 0,
+        microsecond: int = 0,
+        nanosecond: int = 0,
+    ) -> None:
+        super().__init__()
+        self.year = year
+        self.quarter = quarter
+        self.month = month
+        self.week = week
+        self.day = day
+        self.hour = hour
+        self.minute = minute
+        self.second = second
+        self.millisecond = millisecond
+        self.microsecond = microsecond
+        self.nanosecond = nanosecond
+
+    @property
+    def sql(self) -> str:
+        return (
+            f"INTERVAL '{self.year} YEAR, {self.quarter} QUARTER, {self.month} MONTH, "
+            f"{self.week} WEEK, {self.day} DAY, {self.hour} HOUR, {self.minute} MINUTE, "
+            f"{self.second} SECOND, {self.millisecond} MILLISECOND, {self.microsecond} MICROSECOND, "
+            f"{self.nanosecond} NANOSECOND'"
```

Comment:
I just realized that this actually impacts the returned column name. How about not adding the part if it's None (need to use None as the default value above)?

Reply:
@sfc-gh-yixie Changed the constructor and SQL generator to use a dictionary of value mappings so handle this. Also modified the test_column_names.py to not have a year input so check the name matches.